test/test_buffer_protocol.py

```diff
@@ -0,0 +1,168 @@
+import torch.testing._internal.common_utils as common
+from torch.testing._internal.common_device_type import (
+    instantiate_device_type_tests,
+    dtypes, onlyCPU
+)
+from torch.testing import all_types
+
+import torch
+import array
+
+TORCH_TO_ARRAYTYPE = {
+    torch.int8    : 'b',
+    torch.uint8   : 'B',
+    torch.int16   : 'h',
+    torch.int32   : 'i',
+    torch.int64   : 'q',
+    torch.float32 : 'f',
+    torch.float64 : 'd',
+}
+
+def get_dtype_size(dtype):
+    if dtype.is_floating_point:
+        bytes = torch.finfo(dtype).bits / 8
+    else:
+        bytes = torch.iinfo(dtype).bits / 8
+    return int(bytes)
+
+class TestBufferProtocol(common.TestCase):
+    INPUT = [0, 1, 2, 3]
+
+    def _run_test(self, dtype, arr, count=-1, first=0, offset=None, **kwargs):
```

Comment:
Would it be simpler to create NumPy tensors, write them to a buffer, and then compare the result of PyTorch's and NumPy's from_buffer?

Reply:
Do you mean: instead of comparing the resulting PyTorch tensor with a slice of an array, compare it with the result of NumPy's `frombuffer`? If so, computing `last` could be avoided. But, I guess it isn't necessarily simpler.